homeassistant/components/enocean/binary_sensor.py

```diff
@@ -81,36 +88,63 @@ def value_changed(self, packet):
             ['0xf6', '0x10', '0x00', '0x2d', '0xcf', '0x45', '0x30']
         - button released
             ['0xf6', '0x00', '0x00', '0x2d', '0xcf', '0x45', '0x20']
+
+
+        Example packet data for STM 250:
+        - contact closed
+            ['0xd5', '0x9', '0x1', '0x80', '0x5d', '0x5d', '0x0']
+        - contact opened
+            ['0xd5', '0x8', '0x1', '0x80', '0x5d', '0x5d', '0x0']
+        - packet 1 is open/closed
+        - packet 2/3/4/5 is the device address
         """
+
         # Energy Bow
         pushed = None
 
-        if packet.data[6] == 0x30:
-            pushed = 1
-        elif packet.data[6] == 0x20:
-            pushed = 0
-
-        self.schedule_update_ha_state()
-
-        action = packet.data[1]
-        if action == 0x70:
-            self.which = 0
-            self.onoff = 0
-        elif action == 0x50:
-            self.which = 0
-            self.onoff = 1
-        elif action == 0x30:
-            self.which = 1
-            self.onoff = 0
-        elif action == 0x10:
-            self.which = 1
-            self.onoff = 1
-        elif action == 0x37:
-            self.which = 10
-            self.onoff = 0
-        elif action == 0x15:
-            self.which = 10
-            self.onoff = 1
+        match packet.data[0]:
```

Comment:
We need to support Python 3.9. Match / case is only in Python 3.10.

Reply:
Thanks, I have replace the match/case by if/then/else in a new commit.